hexa/metadata/graphql/schema.graphql

```diff
@@ -0,0 +1,107 @@
+"""
+Interface for type implementing metadata
+"""
+interface HasMetadata {
+  extendedId: String!   # Encoded concatenation of the class model type and the instance id
+  metadata: MetadataObject!
+}
+"""
+Generic metadata attribute
+"""
+type MetadataAttribute {
+  id: UUID!
+  key: String!
+  value: JSON
+  system: Boolean!
+}
+
+"""
+Union of all linked to metadata object types
+"""
+union LinkedObject = DatasetVersionFile | DatasetVersion | Dataset
+
+"""
+Generic metadata object
+"""
+type MetadataObject {
```

Comment:
This represents the first version of your PR I guess

Reply:
No, even though I have removed the model, I still kept the dictionary for graphql to return the desired object and extendedId, but it is repetitive and can be deleted